ethstore/src/ethstore.rs

```diff
@@ -509,6 +513,14 @@ impl SimpleSecretStore for EthMultiStore {
 		Err(Error::InvalidPassword)
 	}
 
+	fn agree(&self, account: &StoreAccountRef, password: &str, other: &Public) -> Result<Secret, Error> {
+		let accounts = self.get_matching(account, password)?;
+		for account in accounts {
+			return account.agree(password, other);
```

Comment:
imo, it would be more idiomatic to write

```rust
match accounts.first() {
  Some(ref account) => account.agree(password, other),
  None => Err(Error::InvalidPassword),
}
```